### DISCUSSION TITLE
How can I add Fixed O&M costs in Pypsa?

### DISCUSSION THREAD

**Discussion by AntonisCY97** [PyPSA/PyPSA#1085] [Q&A] [answered]
Hello, I would like to ask if someone has an idea about how to add annual Fixed O&M costs as a percentage of the capital cost of a generator/link/whatever else in a power system optimization in Pypsa?

Thanks in advance

**Comment by e-zaline** [comment]
Hello, I also haven't found how to add fixed costs for existing units, and I would find it very useful.

**Reply by gincrement** [reply, marked as answer]
Fixed O&M is covered in the capital_cost (~ annuity + fixed maintenance).
In the latest version of PyPSA additional fields have been added (overnight_cost, discount_rate, fom_cost) but logic remains the same.

**Comment by FabianHofmann** [comment]
added to more recent pypsa versions as @gincrement pointed out
